service/src/main/java/uk/gov/hmcts/reform/fpl/service/payment/FeeService.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.fpl.service.payment;
+
+import com.google.common.collect.ImmutableList;
+import feign.FeignException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fnp.client.FeesRegisterApi;
+import uk.gov.hmcts.reform.fnp.model.fee.FeeResponse;
+import uk.gov.hmcts.reform.fnp.model.fee.FeeType;
+import uk.gov.hmcts.reform.fpl.config.payment.FeesConfig;
+import uk.gov.hmcts.reform.fpl.config.payment.FeesConfig.FeeParameters;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class FeeService {
+
+    private final FeesConfig feesConfig;
+    private final FeesRegisterApi feesRegisterApi;
+
+    public Optional<FeeResponse> extractFeeToUse(List<FeeResponse> feeResponses) {
+        if (feeResponses == null) {
+            return Optional.empty();
+        }
+
+        return feeResponses.stream()
+            .filter(Objects::nonNull)
+            .max(Comparator.comparing(FeeResponse::getAmount));
+    }
+
+    public List<FeeResponse> getFees(List<FeeType> feeTypes) throws FeignException {
+        if (feeTypes == null) {
+            return ImmutableList.of();
+        }
+
+        return feeTypes.stream()
+            .map(this::makeRequest)
```

Comment:
Don't think so, would be more of an issue that the controller should work around I think. Not sure exactly how it would be handled here rather than the controller handling it to know that it should populate an error

Reply:
Fair enough :+1: resolving